packages/generator/src/edmx-parser/edmx-file-reader.ts

```diff
@@ -57,19 +148,73 @@ function getODataVersion(edmx): ODataVersion {
 function getRoot(edmx) {
   const schema = edmx['edmx:Edmx']['edmx:DataServices'].Schema;
   if (schema.length > 1) {
-    if (schema.length > 2) {
-      throw new Error(
-        'There are more than two schemas in the input metadata file.'
-      );
-    }
     // We assume SFSF edmx files to always have multiple schema tags
     logger.info(`${schema.length} schemas found. Schemas will be merged.`);
-    return schema.reduce(
-      (mergedSchemas, schemaEntry) => ({ ...mergedSchemas, ...schemaEntry }),
-      {}
-    );
+    const ret = mergeSchema(schema);
+    return ret;
   }
-  return schema;
+  const ret2 = mergeSchema(forceArray(schema));
+  return ret2;
+}
+
+function mergeSchema(schema: EdmxMetadataSchema[]) {
```

Comment:
I think this is more than reading. It is parsing and should be done in the edmx parser files.